src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -800,6 +927,9 @@ void ov::npuw::LLMCompiledModel::serialize(std::ostream& stream, const ov::npuw:
         EncryptContext enc_ctx(false, nullptr, nullptr);
         m_kvcache_compiled->serialize(model_stream, enc_ctx);
         m_prefill_compiled->serialize(model_stream, enc_ctx);
+        if (m_tail_mm_compiled_opt.has_value()) {
+            m_tail_mm_compiled_opt.value()->serialize(model_stream, enc_ctx);
+        }
```

Comment:
@smirnov-alexey do you have a generic serialize override for `optional<T>`?
if you do, @AsyaPronina  please use that one.
if you don't, @AsyaPronina here's the place to write your bool (and don't call it `contains_3_models` please).

Reply:
There is in `serialization.cpp`:
```
template <typename T>
void write(std::ostream& stream, const std::optional<T>& var) {
    if (var) {
        write(stream, true);
        write(stream, var.value());
    } else {
        write(stream, false);
    }
}
```

Reply:
Checked with Alexey, this function won't work for std::optional `ov::npuw::CompiledModel`, so either overload is needed for `write` or boolean flag can stay in common `serialize()` method

Reply:
Thats unfortunate but lets keep it as is then.